### ISSUE TITLE
Dateaxis: min max less than 24 hours

### ISSUE THREAD

**Issue by pbtanna17** [amcharts/amcharts4#3100]
**Bug description**

Repro Steps:
1. Create a chart with x-axis as DateAxis
2. Set strictMinMax to true
3. Make sure the difference between min and max is less then 24 hours

Expected Result:
Chart render area to properly divide between the given min and max date.

Actual result:
The chart is slightly skewed towards the right and feels like it almost shows the entire 24 hours with lot of white space before and after the min and max.

Code example:
https://codepen.io/pbtanna17/pen/MWexQbz

I have used one of the existing codepen from amcharts and tweaked some of the params to reproduce the issue.

Amcharts version: amcharts 4.

**Comment by xorspark** [comment]
It looks like the auto-determined `baseInterval` is set to hours, which impacts how to axis is drawn. We generally [recommend manually specifying the `baseInterval`](https://www.amcharts.com/docs/v4/concepts/axes/date-axis/#Manually_specifying_data_granularity) instead of having to chart automatically determine it. In this case, using a minute-based interval seems to fix the problem, e.g.

```
dateAxis.baseInterval = {
  "timeUnit": "minute",
  "count": 10
};
```

**Comment by pbtanna17** [comment]
I have updated the codepen with two data set:
data1: For 4 hours of data
data2: For 24 hours of data
with corresponding min and max in the object.
When we change from data1 to data2, we can see how the chart changes and it takes proper chart space when data2 is applied. It also picks up proper granularity in case of data2. Also, as an experiment remove the last data entry from data2 and the columns width will widen telling me that the granularity is auto populated.

It feels like it happens for data1 as well but the plotting is not correct.

@xorspark : I did try your method before posting this question and yes it does work. Sorry should have mentioned that before but in my case, the granularity is never going to be static. I was very pleased with how amcharts handles that automatically but apparently there is something wrong when it comes to data less than 24 hours.

Any other help or a better solution is appreciated. Thanks.

**Comment by martynasma** [comment]
This seems like a bug, which is not related to granularity. We're looking into it.

**Comment by martynasma** [comment]
Should be fixed in the next version. Will let you know when update is available.

**Comment by pbtanna17** [comment]
@martynasma Thanks a lot for looking into it 👍

**Comment by martynasma** [comment]
Fixed in [4.10.12](https://github.com/amcharts/amcharts4/releases/tag/4.10.12).

## [4.10.12] - 2020-11-26

### Changed
- `DateAxis` setting `timezone` is being deprecated. Please consider setting `timezone` on chart or axis `dateFormatter`. `DateAxis` will now properly place grid in time zones that use non-whole hour fractions, e.g. ("Australia/Adelaide").
- Accessibility: Roles and tabindexes in `ExportMenu` elements has been altered to better comply with ARIA standards.

### Fixed
- Scrollbar could act funky when repeatedly clicking on its grips.
- A minor conflict with Google Charts lib on IE fixed.
- Improved performance of `ValueAxis` scale calculation.
- Setting `min` and `max` on a `DateAxis` could result in excessive overzoom and an empty grid in some cases.
- Overzooming `DateAxis` coul dresult in some extra space shown on it.
- Legend was dynamically adding items on a Treemap chart when drilling-down.
- When Legend's labels' `width` were set in percent, they were not being properly sized.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
